crates/bevy_core/src/time/time.rs

```diff
@@ -34,58 +49,171 @@ impl Time {
     pub(crate) fn update_with_instant(&mut self, instant: Instant) {
         if let Some(last_update) = self.last_update {
             self.delta = instant - last_update;
-            self.delta_seconds_f64 = self.delta.as_secs_f64();
             self.delta_seconds = self.delta.as_secs_f32();
+            self.delta_seconds_f64 = self.delta.as_secs_f64();
+            self.time_since_startup += self.delta.mul_f32(self.relative_speed);
+        } else {
+            self.first_update = Some(instant);
+            self.time_since_startup += (instant - self.startup).mul_f32(self.relative_speed);
         }
 
-        let duration_since_startup = instant - self.startup;
-        self.seconds_since_startup = duration_since_startup.as_secs_f64();
         self.last_update = Some(instant);
     }
 
-    /// The delta between the current tick and last tick as a [`Duration`]
+    /// The delta between the current and last update, as a [`Duration`].
     #[inline]
     pub fn delta(&self) -> Duration {
         self.delta
     }
 
-    /// The delta between the current and last tick as [`f32`] seconds
+    /// The delta between the current and last update, as [`f32`] seconds
     #[inline]
     pub fn delta_seconds(&self) -> f32 {
         self.delta_seconds
     }
 
-    /// The delta between the current and last tick as [`f64`] seconds
+    /// The delta between the current and last update, as [`f64`] seconds.
     #[inline]
     pub fn delta_seconds_f64(&self) -> f64 {
         self.delta_seconds_f64
     }
 
-    /// The time since startup in seconds
+    /// The *virtual* delta between each fixed update step, as a [`Duration`].
+    /// Note that the true delta between fixed update steps varies.
     #[inline]
-    pub fn seconds_since_startup(&self) -> f64 {
-        self.seconds_since_startup
+    pub fn fixed_delta(&self) -> Duration {
+        self.fixed_delta
     }
 
-    /// The [`Instant`] the app was started
+    /// The *virtual* delta between each fixed update step, as [`f32`] seconds.
+    /// Note that the true delta between fixed update steps varies.
     #[inline]
-    pub fn startup(&self) -> Instant {
-        self.startup
+    pub fn fixed_delta_seconds(&self) -> f32 {
+        self.fixed_delta_seconds
+    }
+
+    /// The *virtual* delta between each fixed update step, as [`f64`] seconds.
+    /// Note that the true delta between fixed update steps varies.
+    #[inline]
+    pub fn fixed_delta_seconds_f64(&self) -> f64 {
+        self.fixed_delta_seconds_f64
+    }
+
+    /// Changes [`Time::fixed_delta`]. Users are strongly cautioned against modifying
+    /// the step rate this way, except during startup.
+    /// Prefer [`Time::set_relative_speed`] to ensure that consumers of
+    /// [`Time::fixed_delta`] produce consistent behavior.
+    pub fn set_steps_per_second(&mut self, steps: f32) {
```

Comment:
Why set "steps-per-second" and not "delta time" directly? That sounds contrived.

Reply:
See my first comment.

Reply:
I see the point is trying to avoid misunderstanding, but I'd raise a counterpoint that "steps per second" is like FPS a non-linear scale and makes little sense to setup directly. Also, this is not strictly speaking "steps per second", this is "steps per second on average", which is a bit of a mouthful. Allowing to set `fixed_delta` directly creates a balance with the getter `fixed_delta()` which returns `self.fixed_delta` and not the number of steps per second (on average). Any touching of fixed step simulation parameters should be guarded with strong caution, so I think we can avoid errors with good documentation.

Reply:
Some users being more interested in using Bevy for scientific simulation is probably a reason to consider having a direct `time.set_fixed_delta()`, though.

Reply:
> I don't know what you mean by it's non-linear.

I mean that: https://unrealartoptimization.github.io/book/process/measuring-performance/

> No, a fixed timestep produces and executes an exact number of steps for each second elapsed in-game. 

You're putting an arbitrary limit at 1 second, saying "exactly N steps per second" but "not N/10 steps per 100ms". I say _on average_ over a period sufficiently long, which depends on your workload and stability, you observe N steps per second. But occasionally that might not be the case. This is exactly the argument you made in the other thread when there's a CPU spike and frames are enqueued. Even at smooth frame rate it can't be mathematically exactly N steps per second because steps are discrete and time continuous, and you might get N+1 or N-1 depending on where your "1 second" observation window is. But _on average_ it tends to that yes. I'm just trying to be rigorous in the terminology.

Reply:
Yes, "on average" is a more rigorous definition, but idk it doesn't have the right nuance?

The deviation in measured "steps per second" *at the scale of 1 second* is going to be extremely small. It'll always be very close, since the accumulator closes any gaps.

Reply:
I think talking about averages is weird. Most of the conversation is around the fixed clock which is "perfect" so sometimes referring to the wall clock to clarify might have the opposite effect.

Reply:
> Looking some more I do think that only accepting a frequency feels weird.

Yeah, coming back to this I now agree that having API symmetry here makes more sense.
Would a `Time::set_fixed_delta(delta: Duration)` be enough?

Reply:
Added.